### PR TITLE
Decouple system timer and drivers: SPI

### PR DESCRIPTION
**Pull request by pizi-nordic**
During our work in timer area, we found that system clock frequency is used incorrectly in several drivers.
For example, it is used to obtain bus frequency needed to calculate speed of the SPI bus.
If we select different system timer device, all these calculations become invalid.

This PR fixes the problem by obtaining all needed information directly from the DTS.
Brings us closer to: #15363

### CODE REVIEW

File: dts/arc/emsk.dtsi [hunk 1]
```diff
@@ -147,6 +147,7 @@
 			compatible = "snps,designware-spi";
 			reg = <0xf0006000 0x1000>;
 			label = "SPI_0";
+			clock-frequency = <20000000>;
```
Comment (anangl) [inline dts/arc/emsk.dtsi]: Shouldn't it be `50000000`?
Previously it was specified in `soc/arc/snps_emsk/dts_fixup.h` by:
```
#define DT_SPI_DW_SPI_CLOCK	DT_NS16550_F0009000_CLOCK_FREQUENCY
```
and `clock-frequency` in `uart@f0009000` is set to: `<DT_APB_CLK_HZ>`, and this one is `50000000`.

Actually, I'm not sure if the above definition of `DT_SPI_DW_SPI_CLOCK` is used or rather the one from `soc/arc/snps_emsk/soc.h` but the latter also leads to the value of `MHZ(50)`.

Comment (pizi-nordic) [reply dts/arc/emsk.dtsi]: Fixed, could somebody knowing the hardware confirm correct setting?

**Review by tbursztyka — APPROVED** [review]
seems fine to me, can you restart a CI check?

**Comment by pizi-nordic** [comment]
recheck

**Review by anangl — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]
Don't know too much about the HW. But the code looks clean and I fully support this effort.

**Review by vonhust — APPROVED** [review]
looks ok for me.

**Pull request closed without merging**